clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -126,6 +128,18 @@ public static File tempDirectory(String prefix) throws IOException {
     }
 
     /**
+     * Create a temporary directory named "test" under /temp
+     * @return  the temporary directory just created.
+     */
+    public static File tempDir() {
```

Comment:
> Hmm, not sure about this method:

@ijuma @guozhangwang 

This method was merged from StateTestUtils, but in retrospect I think your comments are correct.  
How about:

- Remove method `tempDir` and add a no-arg `tempDirectory` which delegates to `tempDirectory(null)` 

- Change `tempDirectory(parent, prefix)` to convert IOException to RuntimeException